cpp/include/cudf/column/column_factories.hpp

```diff
@@ -344,20 +345,19 @@ std::unique_ptr<column> make_fixed_width_column(
  *
  * @throws std::bad_alloc if device memory allocation fails
  *
- * @param[in] strings The vector of pointer/size pairs.
- *                Each pointer must be a device memory address or `nullptr`
- * (indicating a null string). The size must be the number of bytes.
+ * @param[in] strings The device span of pointer/size pairs. Each pointer must be a device memory
+   address or `nullptr` (indicating a null string). The size must be the number of bytes.
  * @param[in] stream CUDA stream used for device memory operations and kernel launches.
  * @param[in] mr Device memory resource used for allocation of the column's `null_mask` and children
  * columns' device memory.
  */
 std::unique_ptr<column> make_strings_column(
-  const rmm::device_vector<thrust::pair<const char*, size_type>>& strings,
+  cudf::device_span<thrust::pair<const char*, size_type>> strings,
```

Comment:
I think you want a `const` on the pair type too. 
```suggestion
  cudf::device_span<thrust::pair<const char*, size_type> const> strings,
```

Reply:
Added, thanks.